src/sparql/StardogSparqlParser.ts

```diff
@@ -78,6 +78,114 @@ export class StardogSparqlParser extends BaseSparqlParser {
     ]);
   });
 
+  TriplesSameSubject = this.OVERRIDE_RULE('TriplesSameSubject', () => {
+    this.OR([
+      {
+        ALT: () => {
+          this.SUBRULE(this.VarOrTermOrEmbeddedTriplePattern);
+          this.SUBRULE(this.PropertyListNotEmpty);
+        },
+      },
+      {
+        ALT: () => {
+          this.SUBRULE(this.TriplesNode);
+          this.SUBRULE(this.PropertyList);
+        },
+      },
+    ]);
+  });
+
+  Object = this.OVERRIDE_RULE('Object', () => {
+    this.OR([
+      {
+        ALT: () => this.SUBRULE(this.GraphNode),
+      },
+      {
+        ALT: () => this.SUBRULE(this.EmbeddedTriplePattern),
```

Comment:
Do we need to be aware of using the embedded triples in object lists? 
They can appear in construct queries as follows
```
construct where {
    ?s ?p ?o ;
       ?p2 ?o2, ?o3 .
}
```
where `?o2` and `?o3` are an ObjectList of Objects and can thus be replaced with embedded triple patterns

Reply:
Not totally sure I get your concern here, but `ObjectList` is equivalent to `Object ( ',' Object )*`, so this rule will kick in in that case and allow embedded triple patterns. This is intended, AFAIK (e.g., there's no change to `ObjectList` in [the paper that is the basis for this stuff](https://arxiv.org/pdf/1406.3399.pdf)).

Reply:
I was under the impression that at least in stardog, embedded triples are only valid in the subject position. That is, for `ObjectPath`s (and thus `TriplesBlock`s) at least, there's no support for embedded triples in the object position there, so I'm not sure why in the case of `TriplesTemplate` in `Quads` and `ConstructQuery` it would be.

Reply:
Embedded triples are allowed grammatically in the object position, as in (this is an explicit test case):

```
?x ^:p << ?s :p ?o >> ;
   :q ?o .
```

This is true only for SPARQL. Quoting from internal discussion: "in SPARQL << spo >> is valid in the object position, as simple as that. no need to change anything."

Reply:
Oh, and `ObjectPath`s and `TriplesBlock`s do also support this. `ObjectPath` is equivalent to `GraphNodePath`, which allows embedded triple patterns. And `TriplesBlock`s thereby receive the support, depending on the path through the grammar that is followed.